RIOT-OS/RIOT
# kconfig: introduce migration test in CI

### Contribution description
This PR splits some initial commits from #14055:

- The Murdock script is modified to introduce a test that can be used during the current phase of the migration. The test consists on building certain board/app pairs using Kconfig module modelling to get the list of modules to compile. Then, the binary hash is compared against the hash of the application built normally.

- Some initial modules have been modelled in Kconfig, just the _bare minimum_ to get `examples/hello-world` to build for `samr21-xpro` (this means that, for instance, I did not include the whole STDIO modelling like in #14055).

The idea is to build our way application by application, which will get easier as we model more modules. This test will try to ensure also some synchronization with the current makefile-based dependency resolution system.

The proposed way to do the migration, in order to not interfere with the existing Kconfig files, is to model modules using the `MOD_` prefix, and guard all new symbols with the `TEST_KCONFIG` symbol (which is set when the environmental variable `TEST_KCONFIG` is set). Also, so we don't raise issues with the current `kconfig_features` test, new features (that only exist in Kconfig) have the `HAVE_` prefix for now, which can be changed in the future.

The `TEST_KCONFIG` variable will serve test purposes during migration, it switches from the makefile-based dependency resolution and Kconfig.

**Question**: Some modules like `sys` and `board` have been modelled as non-visible, should they have a prompt to be configurable?

### Testing procedure
- Some things to pay special attention to:
    - Modelling on peripheral drivers
    - The way common configuration files are merged together (via `KCONFIG_ADD_CONFIG` variable) prior to Kconfig running
    - The test itself
- Green CI (check the output for the build of `examples/hello-world` for `samr21-xpro:gnu`, you should see the output of this test)
- Alternatively one could run something like ` /bin/sh -c ". ./.murdock; JOBS=4 compile examples/hello-world samr21-xpro:gnu"`

### Issues/PRs references
Part of #14055
